hazelcast-client/src/main/java/com/hazelcast/client/proxy/NearCachedClientMapProxy.java

```diff
@@ -554,11 +554,13 @@ public void clear() {
     }
 
     @Override
-    protected void onDestroy() {
-        removeNearCacheInvalidationListener();
-        getContext().getNearCacheManager().destroyNearCache(name);
-
-        super.onDestroy();
+    protected void postDestroy() {
+        try {
+            removeNearCacheInvalidationListener();
```

Comment:
Preferring to call supers one first when constructing the object. But here, during destruction, calling supers' destroy later seems ok to me.